tools/data-mapper/Microsoft.Health.Tools.DataMapper/ClientApp/src/components/NavMenu.tsx

```diff
@@ -4,17 +4,60 @@
 // -------------------------------------------------------------------------------------------------
 
 import * as React from 'react';
-import { Navbar, NavbarBrand } from 'reactstrap';
-import { Link } from 'react-router-dom';
+import { Navbar, NavbarBrand, NavbarText } from 'reactstrap';
+import { Link, useLocation } from 'react-router-dom';
+import PersistService from '../services/PersistService';
+import { Mapping } from '../store/Mapping';
+import { default as MappingNameModal, Action } from './mapping/List.Modals.Name';
 import './NavMenu.css';
 
 const NavMenu = (props: {}) => {
+    const [mappingId, setMappingId] = React.useState('');
+    const [mappingTypeName, setMappingTypeName] = React.useState('');
+
+    const location = useLocation();
+
+    React.useEffect(() => {
+        const pathname = location.pathname;
+        const id = pathname.substr(pathname.lastIndexOf('/') + 1);
+        const typeName = PersistService.getMapping(id)?.typeName;
```

Comment:
Just curious - do we expect this (`PersistService.getMapping(id)`) to be empty and if so what is the impact on user experience if we continue with empty typename?

Reply:
Type name is required

Reply:
The root/home page, which contains a list of mappings, does not have a mapping ID in the URL. Since I'm getting the mapping ID from the URL with Line 22, there may be no mapping ID and therefore no mapping or typename, so I use the `?.` operator on Line 23.

Also, with the changes to _PersistService.ts_ in this PR, it's not possible to add a mapping with an empty typename anymore.